File: server/models/Campaign.js
// server/models/Campaign.js
const mongoose = require("mongoose");

const campaignSchema = new mongoose.Schema(
  {
    founderId: { type: String, required: true },
    productName: { type: String, required: true },
    productId: { type: String },
    goal: {
      type: String,
      enum: ["Awareness", "Website visits", "Product sales", "Leads"],
      default: "Product sales"
    },
    targetAudience: { type: String, required: true },
    location: { type: String, default: "Tamil Nadu & South India" },
    budget: { type: Number, default: 500 },
    dailyBudget: { type: Number, default: 250 },
    durationDays: { type: Number, default: 7 },
    status: { type: String, default: "Active" },
    reach: { type: Number, default: 0 },
    relevantAudiencePercent: { type: Number, default: 72 },
    productViews: { type: Number, default: 0 },
    clicks: { type: Number, default: 0 },
    conversions: { type: Number, default: 0 },
    costPerConversion: { type: String, default: "₹18.50" },
    roas: { type: String, default: "3.2x" }
  },
  { timestamps: true }
);

module.exports = mongoose.model("Campaign", campaignSchema);

File: backend/app/schemas/campaign.py
from typing import Optional
from pydantic import BaseModel


class CampaignCreate(BaseModel):
    title: str
    description: Optional[str] = "Targeted D2C Launch Campaign"
    businessId: Optional[str] = None
    mediaUrl: Optional[str] = None
    productId: Optional[str] = None
    campaignType: Optional[str] = "PRODUCT_LAUNCH"
    objective: Optional[str] = "PRODUCT_SALES"
    budget: float = 500.0
    location: Optional[str] = "Tamil Nadu & South India"


class CampaignResponse(BaseModel):
    id: str
    businessId: str
    founderId: str
    title: str
    description: str
    mediaUrl: Optional[str] = None
    campaignType: str
    objective: str
    budget: float
    dailyBudget: float
    durationDays: int
    status: str
    reach: int
    relevantPercent: int
    relevantAudiencePercent: Optional[int] = None
    productViews: int
    clicks: int
    conversions: int


class CampaignAnalyticsResponse(BaseModel):
    campaignId: str
    impressions: int
    views: int
    clicks: int
    engagements: int
    leads: int
    conversions: int
    spend: float
